File: app/src/main/java/com/metehanyl/calarsaat/util/DayUtils.kt
package com.metehanyl.calarsaat.util

import java.util.Calendar

/** Days in display order (Monday first), each paired with its Calendar.DAY_OF_WEEK value. */
object DayUtils {

    val orderedDays: List<Pair<Int, String>> = listOf(
        Calendar.MONDAY to "Pzt",
        Calendar.TUESDAY to "Sal",
        Calendar.WEDNESDAY to "Çar",
        Calendar.THURSDAY to "Per",
        Calendar.FRIDAY to "Cum",
        Calendar.SATURDAY to "Cmt",
        Calendar.SUNDAY to "Paz"
    )

    fun summarize(days: Set<Int>): String {
        if (days.isEmpty()) return "Bir kez"
        if (days.size == 7) return "Her gün"
        val weekdays = setOf(
            Calendar.MONDAY, Calendar.TUESDAY, Calendar.WEDNESDAY,
            Calendar.THURSDAY, Calendar.FRIDAY
        )
        val weekend = setOf(Calendar.SATURDAY, Calendar.SUNDAY)
        if (days == weekdays) return "Hafta içi"
        if (days == weekend) return "Hafta sonu"
        return orderedDays.filter { it.first in days }.joinToString(", ") { it.second }
    }
}

File: ios/CalarSaat/Util/DayUtils.swift
import Foundation

/// Maps `Calendar.Component.weekday` (1=Sun...7=Sat) to Turkish display labels.
/// Same convention as `EKCalendar`/`DateComponents.weekday`, so no conversion
/// from Android's `Calendar.DAY_OF_WEEK` is needed.
enum DayUtils {
    static let orderedDays: [(Int, String)] = [
        (2, "Pzt"), (3, "Sal"), (4, "Çar"), (5, "Per"), (6, "Cum"), (7, "Cmt"), (1, "Paz"),
    ]

    static func summarize(_ days: Set<Int>) -> String {
        if days.isEmpty { return "Bir kez" }
        if days.count == 7 { return "Her gün" }
        let weekdays: Set<Int> = [2, 3, 4, 5, 6]
        let weekend: Set<Int> = [1, 7]
        if days == weekdays { return "Hafta içi" }
        if days == weekend { return "Hafta sonu" }
        return orderedDays.filter { days.contains($0.0) }.map(\.1).joined(separator: ", ")
    }
}
